test/dash/dash_parser_manifest_unit.js

```diff
@@ -1566,39 +1569,44 @@ describe('DashParser Manifest', () => {
     ].join('\n');
 
     fakeNetEngine.setResponseText('dummy://foo', manifestText);
-    const config = shaka.util.PlayerConfiguration.createDefault().manifest;
-    config.dash.ignoreMinBufferTime = true;
     parser.configure(config);
+    return parser.start('dummy://foo', playerInterface);
+  }
 
+  async function getMaxSegmentDuration(ignore) {
+    const config = shaka.util.PlayerConfiguration.createDefault().manifest;
+    config.dash.ignoreMaxSegmentDuration = ignore;
     /** @type {shaka.extern.Manifest} */
-    const manifest = await parser.start('dummy://foo', playerInterface);
-    const minBufferTime = manifest.minBufferTime;
+    const manifest = await getParsedManifest(config);
+    return manifest.presentationTimeline.getMaxSegmentDuration();
+  }
+
+  async function getMinBufferTime(ignore) {
+    const config = shaka.util.PlayerConfiguration.createDefault().manifest;
+    config.dash.ignoreMinBufferTime = ignore;
+    /** @type {shaka.extern.Manifest} */
+    const manifest = await getParsedManifest(config);
+    return manifest.minBufferTime;
+  }
+
+  it('override manifest value if ignoreMinBufferTime is true', async () => {
+    const minBufferTime = await getMinBufferTime(true);
     expect(minBufferTime).toBe(0);
   });
 
   it('get manifest value if ignoreMinBufferTime is false', async () => {
-    const manifestText = [
-      '<MPD minBufferTime="PT75S">',
-      '  <Period id="1" duration="PT30S">',
-      '    <AdaptationSet id="1" mimeType="video/mp4">',
-      '      <Representation id="video-sd" width="640" height="480">',
-      '        <BaseURL>v-sd.mp4</BaseURL>',
-      '        <SegmentBase indexRange="100-200" />',
-      '      </Representation>',
-      '    </AdaptationSet>',
-      '  </Period>',
-      '</MPD>',
-    ].join('\n');
+    const minBufferTime = await getMinBufferTime(false);
+    expect(minBufferTime).toBe(75);
```

Comment:
I appreciate that you're trying to follow the "Don't Repeat Yourself" principle by factoring out the manifest text, but even though you've made the tests shorter and simpler, you've made them harder to understand.

When you read this test, it's totally unclear why 75 is the correct value, because you can't see the manifest text.  You can't even see it by going to getMinBufferTime, because the manifest is one level deeper in getParsedManifest.

There are some other approaches you could take, like putting shared manifest text into a variable with a meaningful name.  For example, this test would be easier to understand, in my opinion:

```js
    const minBufferTime = await getMinBufferTime(
        /* manifestText= */ manifestWithMinBuffer75,
        /* ignoreMinBufferTime= */ false);
    expect(minBufferTime).toBe(75);
```

That way, even with the bulky manifest text factored out, you know exactly why this is the correct expectation.  It also makes it difficult for someone to change that shared text in a way that breaks the test.  If I see a variable with MinBufferTime75 in the name, I won't change the minBufferTime value in it to anything else.

Or, if you don't like that suggestion, you could always just put these back the way they were with explicit text in each test.

Or, come up with another idea and pitch it.  We can be flexible.